staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -379,9 +379,17 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	localError := make(chan struct{})
 	remoteDone := make(chan struct{})
 
+	drainRemoteStream := func() {
+		_, err := io.ReadAll(dataStream)
+		if err != nil {
+			fmt.Fprintf(pf.out, "error draining all remain data from remote stream after connection broken\n")
+		}
+	}
+
 	go func() {
 		// Copy from the remote side to the local port.
 		if _, err := io.Copy(conn, dataStream); err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
if this error came from reading from dataStream, then trying to read again in drainRemoteStream will just repeat and double-log the error, right? that doesn't seem correct

Reply:
There are two situations where the error here can be triggered: 
1. Reading from `dataStream`, in which case `drainRemoteStream` will indeed log the error again.
2. Writing to `conn`, at this point dataStream can be read normally, and the goal of this PR is to discard any remaining data in `dataStream` during this situation.

Reply:
Or can we ignore the error generated by `io.Copy(io.Discard, dataStream)`? In my opinion, it is indeed not important because in any case, we have already logged the corresponding error.

Reply:
I sort of think we should ignore any drain error when we already logged there was an error.

I also sort of think we should always drain even if the error is one we're not going to log because it's an "expected" network closed error (since we don't know which side that error came from).

Maybe something like this:

```go
if _, err := io.Copy(conn, dataStream); err != nil {
	if !strings.Contains(err.Error(), "use of closed network connection") {
		runtime.HandleError(fmt.Errorf("error copying from remote stream to local connection: %v", err))
	}
	// drain the remote datastream to make sure we don't leave it blocked
	_, err := io.Copy(io.Discard, dataStream)
}
```

Reply:
LGTM, accepted your suggestion.